File: src/types/act.ts
export interface ActServiceItem {
  id: string;
  code: string;
  name: string;               // Наименование услуги
  defaultPrice: number;       // Дефолтная цена из ТЗ
  price: number;              // Фактическая цена (редактируемая)
  quantity: number;           // Количество (редактируемое)
  amount: number;             // Сумма = price * quantity
  enabled: boolean;           // Включена ли позиция в текущий акт (чекбокс)
  isCustom?: boolean;         // Кастомная добавленная услуга
}

export interface ActExecutorRequisites {
  companyName: string;
  legalAddress: string;
  innKpp: string;
  checkingAccount: string;
  corrAccount: string;
  bankName: string;
  bik: string;
  corrBank: string;
  corrBankBik: string;
  corrBankKs: string;
  swiftCode: string;
}

export interface Act {
  id: string;
  actNumber: string;          // Номер акта (например: АКТ-2026-0805-01)
  shipmentId?: string;
  shipmentNumber?: string;
  date: string;               // Дата акта (YYYY-MM-DD)
  operatorName: string;       // ФИО оператора
  clientId: string;
  clientName: string;
  clientRequisitesText: string; // Реквизиты контрагента
  executorRequisites: ActExecutorRequisites;
  items: ActServiceItem[];
  totalSum: number;           // Итоговая сумма акта
  status: 'draft' | 'signed' | 'paid';
  createdAt: string;
  updatedAt: string;
}

export type CreateActDto = Omit<Act, 'id' | 'createdAt' | 'updatedAt'>;

File: backend/app/schemas/act.py
from pydantic import BaseModel
from typing import Optional, List, Any
from datetime import datetime

class ActServiceItemSchema(BaseModel):
    id: Optional[str] = None
    code: str
    name: str
    price: float
    quantity: int
    amount: float
    enabled: bool
    isCustom: Optional[bool] = False

class ActExecutorRequisitesSchema(BaseModel):
    companyName: str
    legalAddress: str
    innKpp: str
    checkingAccount: str
    corrAccount: str
    bankName: str
    bik: str
    corrBank: Optional[str] = None
    corrBankBik: Optional[str] = None
    corrBankKs: Optional[str] = None
    swiftCode: Optional[str] = None

class ActCreate(BaseModel):
    actNumber: str
    shipmentId: Optional[str] = None
    shipmentNumber: Optional[str] = None
    date: str
    operatorName: str
    clientId: str
    clientName: str
    clientRequisitesText: str
    executorRequisites: ActExecutorRequisitesSchema
    items: List[ActServiceItemSchema]
    totalSum: float
    status: Optional[str] = "signed"

class ActResponse(ActCreate):
    id: str
    createdAt: Optional[datetime] = None
    updatedAt: Optional[datetime] = None

    class Config:
        from_attributes = True
